Source/WebCore/loader/TextResourceDecoder.cpp

```diff
@@ -330,8 +330,10 @@ static inline bool shouldPrependBOM(std::span<const uint8_t> data)
 String TextResourceDecoder::textFromUTF8(std::span<const uint8_t> data)
 {
     auto decoder = TextResourceDecoder::create("text/plain"_s, "UTF-8");
-    if (shouldPrependBOM(data))
-        decoder->decode("\xef\xbb\xbf", 3);
+    if (shouldPrependBOM(data)) {
+        constexpr std::array<uint8_t, 3> bom = { 0xEF, 0xBB, 0xBF };
```

Comment:
I am pretty sure `std::array` can deduce template arguments if you omit them. It might be hard since these byte values are `int`, not `uint8_t`, but we may be able to get the `3` deduced if we change the syntax to construction rather than initialization style. Probably not worth it, just a stray thought.

Reply:
I'll try to find a solution but at the moment, I don't know how to do it besides:
```
constexpr std::array bom = { static_cast<uint8_t>(0xEF), static_cast<uint8_t>(0xBB), static_cast<uint8_t>(0xBF) };
```

There is `std::to_array` too but I'm not sure we can use it yet. I can try to see how the bots like it.

Reply:
`constexpr auto bom = std::to_array<uint8_t>({ 0xEF, 0xBB, 0xBF });` builds locally. Let's see what EWS says!

Reply:
Excellent. I would not have found it; instead I saw `std::make_array`, which is not in the standard yet.